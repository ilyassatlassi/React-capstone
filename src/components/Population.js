import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchPopulationData,
  selectFilteredData,
} from '../redux/stock/populationSlice';
import './style/Population.css';

import CountryCard from './CountryCard';

export default function Population({ searchTerm }) {
  const dispatch = useDispatch();
  const filteredData = useSelector(selectFilteredData);
  const status = useSelector((state) => state.population.status);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchPopulationData());
    }
    // eslint-disable-next-line
  }, [dispatch]);

  return (
    <div className="country-list">
      {filteredData
        .filter((country) => country.name.toLowerCase().includes(searchTerm.toLowerCase()))
        .slice(0, 100)
        .map((country) => (
          <CountryCard key={country.name} country={country} />
        ))}
    </div>
  );
}
