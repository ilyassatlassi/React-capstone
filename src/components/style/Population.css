.container {
    position: relative;
    margin-top: 10px;
    overflow: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 40px;
    width: 100%;
    padding: 10px;
    border-radius: 30px;
  }

  .country-list {
    display: grid;
    /* grid-auto-columns: 1fr; */
    grid-template-columns: 1fr 1fr;
    gap: 5px 5px;
    width: 100%;
  }
.country-list a {
text-decoration: none;
}