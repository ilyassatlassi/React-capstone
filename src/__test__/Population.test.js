import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Population from '../components/Population';

const mockStore = configureMockStore();

describe('Population component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      population: {
        data: [
          {
            name: 'Country 1',
            population: 1000,
            flag: 'https://restcountries.com/data/afg.svg',
            capital: 'Capital 1',
            region: 'Region 1',
            subregion: 'Subregion 1',
          },
          {
            name: 'Country 2',
            population: 2000,
            flag: 'https://restcountries.com/data/ala.svg',
            capital: 'Capital 2',
            region: 'Region 2',
            subregion: 'Subregion 2',
          },
        ],
        status: 'succeeded',
        error: null,
      },
    });
  });

  it('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Population searchTerm="" />
        </BrowserRouter>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
