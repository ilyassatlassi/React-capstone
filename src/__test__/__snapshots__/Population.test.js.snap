// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Population component renders correctly 1`] = `
<div
  class="country-list"
>
  <a
    href="/details/Country 1"
  >
    <div
      class="country-card"
    >
      <img
        alt="Country 1 flag"
        class="country-flag"
        src="https://restcountries.com/data/afg.svg"
      />
      <h2>
        Country 1
      </h2>
      <p>
        Population:
        1000
      </p>
    </div>
  </a>
  <a
    href="/details/Country 2"
  >
    <div
      class="country-card"
    >
      <img
        alt="Country 2 flag"
        class="country-flag"
        src="https://restcountries.com/data/ala.svg"
      />
      <h2>
        Country 2
      </h2>
      <p>
        Population:
        2000
      </p>
    </div>
  </a>
</div>
`;
